akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2FrameProbe.scala

```diff
@@ -173,6 +175,13 @@ private[http] object Http2FrameProbe extends Matchers {
         (lastStreamId, ErrorCode.byId(reader.readIntBE()))
       }
 
+      override def expectSETTINGS(): Unit = {
+        // (6.5) The stream identifier for a SETTINGS frame MUST be zero (0x0).
+        // Ignore the payload for the time being. SETTINGS frames
+        // should have an empty payload anyway.
```

Comment:
No, the payload contains the settings. Let's either return the payload as a ByteString as suggested before or use `Http2FrameParsing` to actually parse the settings.

Reply:
(I already added an `expectFrame` that returns the parsed `FrameEvent` that you can then pattern-match on. perhaps it'd be neat to give that a generic parameter so `expectFrame[SettingsFrame]()` throws when another type of frame is encountered, and returns the frame in the correct type right away. Anyway, not necessarily for this PR :) )

Reply:
I think there's a bit of a mess in the methods we have in the probe. For example, the new [`expectFrame()`](https://github.com/akka/akka-http/blob/master/akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2FrameProbe.scala#L113) is useful to expect a frame but, unlike other `expectXyz` methods it doesn't validate any field on the frame (e.g. `expectDATAFrame(streamId: Int)` validates the `streamId` is the value expected).

I think we have an opportunity to review these methods and, for example, add the generic param @raboof mentioned above.

I'd rather do it on a separate PR.

Reply:
> Let's either return the payload as a ByteString as suggested before or use Http2FrameParsing to actually parse the settings.

+1 I've improved it to return frame.